src/AbstractHyperrectangle.jl

```diff
@@ -181,3 +182,32 @@ function ∈(x::AbstractVector{N},
     end
     return true
 end
+
+"""
+    linear_map(M::AbstractMatrix, H::AbstractHyperrectangle)
+
+Concrete linear map of an abstract hyperrectangle.
+
+### Input
+
+- `M` -- matrix
+- `S` -- abstract hyperrectangle
+
+### Output
+
+The polytope in V-representation obtained by applying the linear map ``M`` to
+the set ``H``. If the given hyperrectangle is two-dimensional, a polygon instead
+of a general polytope is returned. 
+"""
+function linear_map(M::AbstractMatrix, H::AbstractHyperrectangle{N}) where {N}
+    if dim(H) == 2
+        T = VPolygon
+    else
+        T = VPolytope
+    end
+    vlist = Vector{Vector{N}}()
```

Comment:
For the moment it would be more efficient to use the size of `vertices_list` for initialization. But once we have the iterator (instead of the list) available, we may replace it anyway, so we can also just keep it.

Reply:
> For the moment it would be more efficient to use the size of vertices_list for initialization.

But `vertices_list` does the actual computation, ie. if i do `sizehint!(vlist, length(vertices_list(H))` it will already compute the list of vertices. Or you have sth else in mind?

Reply:
You call `vertices_list` below. Just call it above and cache this value.